test/lib/formatPercentage.test.ts

```diff
@@ -0,0 +1,17 @@
+import { describe, it, expect } from "vitest";
+import { formatPercentageToFirstNonZeroDigit } from "@/lib/formatPercentage";
+
+describe("formatPercentage", () => {
+  it("should round to first non-zero digit", () => {
+    expect(formatPercentageToFirstNonZeroDigit(97.5)).to.eq("97.5");
+    expect(formatPercentageToFirstNonZeroDigit(97.001)).to.eq("97");
+    expect(formatPercentageToFirstNonZeroDigit(97.01)).to.eq("97");
+    expect(formatPercentageToFirstNonZeroDigit(1.008)).to.eq("1");
+    expect(formatPercentageToFirstNonZeroDigit(1.99)).to.eq("2");
+    expect(formatPercentageToFirstNonZeroDigit(0.99)).to.eq("0.9");
```

Comment:
Love the tests! <3

I know this would complicate your algorithm, but wouldn't it be better to round up or down depending on the last digit in cases like this one? 0.99 is so close to 1 that it should probably shown as such.

Also, I think you should have one example of an integer and test that it doesn't change the integer at all.

```
    expect(formatPercentageToFirstNonZeroDigit(97)).to.eq("97");
```

Reply:
This is obviously non-blocking. This would perhaps be a neat little improvement.

Reply:
Actually ended up implementing this. I added a testcase, which uncovered a bug (it switched to scientific notation because the number was very small), and as a result of fixing that bug it ended up working as described above.